runtime/tests/runtime_integration_tests/test_revoke_exported_cdi_handle.rs

```diff
@@ -0,0 +1,142 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_api::{mailbox::RevokeExportedCdiHandleReq, SocManager};
+use caliptra_common::mailbox_api::{CommandId, MailboxReq, MailboxReqHeader};
+use caliptra_error::CaliptraError;
+use caliptra_hw_model::HwModel;
+use caliptra_runtime::RtBootStatus;
+use dpe::{
+    commands::{Command, DeriveContextCmd, DeriveContextFlags},
+    context::ContextHandle,
+    response::Response,
+    DPE_PROFILE,
+};
+
+use crate::common::{assert_error, execute_dpe_cmd, run_rt_test, DpeResult, RuntimeTestArgs};
+
+#[test]
+fn test_revoke_exported_cdi_handle() {
+    let mut model = run_rt_test(RuntimeTestArgs::default());
+    model.step_until(|m| {
+        m.soc_ifc().cptra_boot_status().read() == u32::from(RtBootStatus::RtReadyForCommands)
+    });
+
+    let export_cdi_cmd = DeriveContextCmd {
+        handle: ContextHandle::default(),
+        data: [0; DPE_PROFILE.get_tci_size()],
+        flags: DeriveContextFlags::EXPORT_CDI | DeriveContextFlags::CREATE_CERTIFICATE,
+        tci_type: 0,
+        target_locality: 0,
+    };
+
+    let Some(Response::DeriveContextExportedCdi(original_cdi_resp)) = execute_dpe_cmd(
+        &mut model,
+        &mut Command::DeriveContext(&export_cdi_cmd),
+        DpeResult::Success,
+    ) else { panic!("expected derive context resp!") };
+
+    let mut cmd = MailboxReq::RevokeExportedCdiHandle(RevokeExportedCdiHandleReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        exported_cdi_handle: original_cdi_resp.exported_cdi,
+    });
+    cmd.populate_chksum().unwrap();
+
+    model
+        .mailbox_execute(
+            CommandId::REVOKE_EXPORTED_CDI_HANDLE.into(),
+            cmd.as_bytes().unwrap(),
+        )
+        .expect("Expected REVOKE_EXPORTED_CDI_HANDLE to pass");
```

Comment:
nit: Might be worth a test that calls export a second time after the revoke. Although I think that's covered in the libcaliptra tests, so maybe it's redundant.

Reply:
Added a test case for this.